yazi-plugin/preset/plugins/extract.lua

```diff
@@ -47,23 +57,45 @@ function M:try_with(url, pwd)
 		fail("Failed to determine a temporary directory for %s", url)
 	end
 
+	os.execute("mkdir -p " .. tostring(tmp))
 	local archive = require("archive")
-	local child, code = archive:spawn_7z { "x", "-aou", "-p" .. pwd, "-o" .. tostring(tmp), tostring(url) }
-	if not child then
-		fail("Spawn `7z` and `7zz` both commands failed, error code %s", code)
-	end
+	if self:is_tar_compatible(tostring(url)) then
+		local child, code = archive:spawn_tar { "-xvf", tostring(url), "-C", tostring(tmp) }
+		if not child then
+			fail("Failed to spawn `tar` command, error code %s", code)
+		end
 
-	local output, err = child:wait_with_output()
-	if output and output.status.code == 2 and archive:is_encrypted(output.stderr) then
-		fs.remove("dir_clean", tmp)
-		return true -- Needs retry
-	end
+		local output, err = child:wait_with_output()
+		if output and output.status.code == 2 and archive:is_encrypted(output.stderr) then
```

Comment:
`archive:is_encrypted(output.stderr) `, Can `tar` be encrypted?

Reply:
Tar does not have any built-in encryption like 7z, needs to be used with gpg or other tools.